docs/api/dest.md

```diff
@@ -0,0 +1,122 @@
+<!-- front-matter
+id: api-dest
+title: dest()
+hide_title: true
+sidebar_label: dest()
+-->
+
+# dest()
+
+Creates a stream for writing [Vinyl][vinyl-concepts] objects to the file system.
+
+## Usage
+
+```js
+const { src, dest } = require('gulp');
+
+function copy() {
+  return src('input/*.js')
+    .pipe(dest('output/'));
+}
+
+exports.copy = copy;
+```
+
+## Signature
+
+```js
+dest(directory, [options])
+```
+
+### Parameters
+
+| parameter | type | note |
+|:--------------:|:-----:|--------|
+| directory <br> **(required)** | string <br> function | The path of the output directory where files will be written. If a function is used, the function will be called with each Vinyl object and must return a string directory path. |
+| options | object | Detailed in [Options][options-section] below. |
+
+### Returns
+
+A stream that can be used in the middle or at the end of a pipeline to create files on the file system.
+Whenever a Vinyl object is passed through the stream, it writes the contents and other details out to the file system at the given directory. If the Vinyl object has a `symlink` property, a symbolic link will be created instead of writing the contents. After the file is created, its [metadata will be updated][metadata-updates-section] to match the Vinyl object.
+
+Whenever a file is created on the file system, the Vinyl object will be modified.
+* The `cwd`, `base`, and `path` properties will be updated to match the created file.
+* The `stat` property will be updated to match the file on the file system.
+* If the `contents` property is a stream, it will be reset so it can be read again.
+
+### Errors
+
+When `directory` is an empty string, throws an error with the message, "Invalid dest() folder argument. Please specify a non-empty string or a function."
+
+When `directory` is not a string or function, throws an error with the message, "Invalid dest() folder argument. Please specify a non-empty string or a function."
+
+When `directory` is a function that returns an empty string or `undefined`, emits an error with the message, "Invalid output folder".
+
+### Options
+
+
+**For options that accept a function, the passed function will be called with each Vinyl object and must return a value of another listed type.**
+
+| name | type | default | note |
+|:-------:|:------:|-----------|-------|
+| cwd | string <br> function | `process.cwd()` | The directory that will be combined with any relative path to form an absolute path. Is ignored for absolute paths. Use to avoid combining `directory` with `path.join()`. |
+| mode | number <br> function | `stat.mode` of the Vinyl object | The mode used when creating files. If not set and `stat.mode` is missing, the process' mode will be used instead. |
+| dirMode | number <br> function | | The mode used when creating directories. If not set, the process' mode will be used. |
```

Comment:
Same as above, except for directories the default mode is defined in another module, fs-mkdirp-stream.

Reply:
Same as above